### PR TITLE
Create docs python update

### PR DESCRIPTION
**Pull request by KanclerzPiotr**
# Existing Sample Changes
## Description

Update of script for generating docs

Added support for Linux
Refactor for increased readability
Added check if License.txt exists
Added check if README.md exists
If new sample is added, new record in guids.json is added
Added informations about deleted samples to main README
CHANGELOG.md is now sorted by version descending and by name ascending

## External Dependencies
None

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [X] Command Line
- [ ] oneapi-cli
- [ ] Visual Studio
- [ ] Eclipse IDE
- [ ] VSCode
- [ ] When compiling the compliler flag "-Wall -Wformat-security -Werror=format-security" was used

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,18 +82,25 @@ See [CONTRIBUTING wiki](https://github.com/oneapi-src/oneAPI-samples/blob/master
...
 |2021.4.0|[Intel® Python Scikit-learn Extension Getting Started](https://github.com/oneapi-src/oneAPI-samples/tree/master/AI-and-Analytics/Getting-Started-Samples/Intel_Extension_For_SKLearn_GettingStarted)|This sample illustrates how to do Image classification using SVM classifier from Python API package SKlearnex with the use of Intel® oneAPI Data Analytics Library (oneDAL).|
 |2021.4.0|[Merge Sort](https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/DPC++FPGA/ReferenceDesigns/merge_sort)|A Reference design demonstrating merge sort on an Intel® FPGA|
 |2021.4.0|[Private Copies](https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/DPC++FPGA/Tutorials/Features/private_copies)|An Intel® FPGA tutorial demonstrating how to use the private_copies attribute to trade off the resource use and the throughput of a DPC++ FPGA program|
 |2021.4.0|[Stall Enable](https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/DPC++FPGA/Tutorials/Features/stall_enable)|An Intel® FPGA tutorial demonstrating the use_stall_enable_clusters attribute|
 
 Total Samples: 158
 
+
+## Deleted Code Samples
```
Comment (JoeOster) [inline README.md:94]: This is showing as 2021..Gold, when in fact I believe it was 2021.1.Gold-2021.3.  We should also show the path if we are going to suggest a previous version  DirectProgramming/DPC++FPGA/Tutorials/Tools/use_library/CMakeLists.txt and the previous version should include the hyperlink to its GitHub tag aka https://github.com/oneapi-src/oneAPI-samples/releases/tag/2021.3.0

Please use 2022.3, should read as 2021.3

File: CODESAMPLESLIST.md [hunk 2]
```diff
@@ -157,13 +158,10 @@ This Page is a complete list of oneAPI Code Samples, sorted by Alphabetical orde
 |[Tutorials](https://github.com/oneapi-src/oneAPI-samples/tree/master/Libraries/oneCCL/tutorials)|['CPU', 'GPU']|Intel® oneCCL Tutorials|
```
Comment (JoeOster) [inline CODESAMPLESLIST.md:158]: Insufficient sample name, this is an issue with the samples json, not the code provided here

The json needs updating to be updated with a clear title, something like **Tutorials: oneCCL Getting started** - check with Louie Tsai for recommended change

File: CODESAMPLESLIST.md [hunk 3]
```diff
@@ -157,13 +158,10 @@ This Page is a complete list of oneAPI Code Samples, sorted by Alphabetical orde
...
 |[Tutorials](https://github.com/oneapi-src/oneAPI-samples/tree/master/Libraries/oneDNN/tutorials)|['CPU', 'GPU']|Intel® oneDNN Tutorials|
```
Comment (JoeOster) [inline CODESAMPLESLIST.md:159]: Insufficient sample name, this is an issue with the samples json, not the code provided here

The json needs updating to be updated with a clear title, something like **Tutorials: oneDNN Getting started** - check with Louie Tsai for recommended change

File: CHANGELOGS.md [hunk 4]
```diff
@@ -26,142 +23,144 @@ This document shows the history of when a specific sample was introduced to the
...
-|2021.1.Gold|[DPC Reduce](https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/DPC++/ParallelPatterns/dpc_reduce)|This sample models transform Reduce in different ways showing capability of Intel® oneAPI|
+|2021.1.Gold|[AWS Pub Sub](https://github.com/oneapi-src/oneAPI-samples/tree/master/Tools/IoTConnectionTools/aws-pub-sub)|This sample uses the Message Broker for AWS* IoT to send and receive messages through an MQTT connection|
+|2021.1.Gold|[Zero Copy Data Transfer](https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/zero_copy_data_transfer)|An Intel® FPGA tutorial demonstrating zero-copy host memory using the SYCL restricted Unified Shared Memory (USM) model|
+|2021.1.Gold|[Base: Vector Add](https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/DPC++/DenseLinearAlgebra/vector-add)|This simple sample adds two large vectors in parallel. Provides a ‘Hello World!’ like sample to ensure your environment is setup correctly using simple Intel® oneAPI Data Parallel C++ (DPC++)|
```
Comment (JoeOster) [inline CHANGELOGS.md:155]: Why are the last 9 out of order?

**Review by JoeOster — CHANGES_REQUESTED** [review]
CRR json is insufficient as a name, check with Mike Tucker for Guidance

Matrix Mul is used in two places as a name, please fix it to Matrix Multiply ...

File: CHANGELOGS.md (hunk 4 above)
Comment (KanclerzPiotr) [reply CHANGELOGS.md:155]: There was an error in guids.json - lack of first letters in the name

File: CODESAMPLESLIST.md (hunk 2 above)
Comment (KanclerzPiotr) [reply CODESAMPLESLIST.md:158]: I think that this name is good enough, it is not one sample but it should be list of jupyter notebooks.

File: CODESAMPLESLIST.md (hunk 3 above)
Comment (KanclerzPiotr) [reply CODESAMPLESLIST.md:159]: I think that this name is good enough, it is not one sample but list of jupyter notebooks,

**Comment by JoeOster** [comment]
@mdbtucker - Please verify these errors for FPGA are expected for 2021.3

**Comment by mdbtucker** [comment]
> @mdbtucker - Please verify these errors for FPGA are expected for 2021.3

Confirmed

File: CODESAMPLESLIST.md [hunk 5]
```diff
@@ -157,13 +158,10 @@ This Page is a complete list of oneAPI Code Samples, sorted by Alphabetical orde
...
 |[UP2 LEDS](https://github.com/oneapi-src/oneAPI-samples/tree/master/Tools/IoTConnectionTools/up2-leds)|['CPU']|This sample shows how to use the LED class and APIs of the Eclipse* MRAA library. It is intended to run on the UP Squared board, and will utilize the 4 built-in color LEDs located under the Ethernet ports. No additional hardware is required|
-|[Use Library](https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/DPC++FPGA/Tutorials/Tools/use_library)|['FPGA']|An Intel® FPGA Tutorial demonstrating how to create Intel® FPGA libraries and to incorporate them in a Intel® oneAPI Data Parallel C++ (DPC++) project|
```
Comment (JoeOster) [inline CODESAMPLESLIST.md:160]: @mdbtucker-any thoughts on maybe a better name in the json?

File: CODESAMPLESLIST.md [hunk 6]
```diff
@@ -157,13 +158,10 @@ This Page is a complete list of oneAPI Code Samples, sorted by Alphabetical orde
...
 |[Vector Add DPCT](https://github.com/oneapi-src/oneAPI-samples/tree/master/Tools/Migration/vector-add-dpct)|['CPU']|Simple project to illustrate the basic migration of CUDA code. Use this sample to ensure your environment is configured correctly and to understand the basics of migrating existing CUDA projects to Intel® oneAPI Data Parallel C++ (DPC++)|
 |[Vectorize VecMatMult](https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/Fortran/DenseLinearAlgebra/vectorize-vecmatmult)|['CPU']|Fortran Tutorial - Using Auto Vectorization|
 |[Zero Copy Data Transfer](https://github.com/oneapi-src/oneAPI-samples/tree/master/DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/zero_copy_data_transfer)|['FPGA']|An Intel® FPGA tutorial demonstrating zero-copy host memory using the SYCL restricted Unified Shared Memory (USM) model|
 |[oneCCL Getting Started](https://github.com/oneapi-src/oneAPI-samples/tree/master/Libraries/oneCCL/oneCCL_Getting_Started)|['CPU', 'GPU']|Basic Intel® oneCCL programming model for both Intel® CPU and GPU|
```
Comment (JoeOster) [inline CODESAMPLESLIST.md:164]: seems weird this got sorted to the bottom

**Review by JoeOster — CHANGES_REQUESTED** [review]

File: CODESAMPLESLIST.md (hunk 6 above)
Comment (KanclerzPiotr) [reply CODESAMPLESLIST.md]: sorting was case insensitive,

**Comment by JoeOster** [comment]
@KanclerzPiotr - There were a couple of last-minute changes for 2021.4 (master branch) those changes were later pushed to development. This PR is messy and needs to reflect only those changes you are making for clarity.

When you rebase, you need to rebase to the appropriate branch, in the past, you probably rebased to master, now it needs to be to the development branch.
One way to do this:
git checkout development
git pull origin development **-> this fetches the remote version of the development branch and merges it (or rebases it, depending on your pull strategy) into/onto your local branch. This way the local and the remote versions of development are the same**
git checkout <your branch>
git rebase develop

@mdbtucker may have other input as well on this subject

File: CODESAMPLESLIST.md (hunk 5 above)
Comment (KanclerzPiotr) [reply CODESAMPLESLIST.md]: @mdbtucker could you check it?

Comment (mdbtucker) [reply CODESAMPLESLIST.md]: The change to the CRR name (to CRR Binomial Tree) in the JSON looks fine to me.

**Review by JoeOster — APPROVED** [review]

**Pull request merged**